### PR TITLE
Signed/Approval message with aadhaar QR

### PR DESCRIPTION
**Pull request by vuvoth**
## Motivation
#133 

## Change Summary

Add dummy constraint signedMessage * signedMessaage = messageSquare; 

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [ ] PR title adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [ ] PR has a [change set](https://github.com/privacy-scaling-explorations/anon-aadhaar/CHANGELOG.md)
- [ ] PR has been tagged with a change label(s) (i.e. documentation, feature, bug fix, or chore)
- [ ] PR includes documentation if necessary.

## Additional Context

If this is a relatively large or complex change, provide more details here that will help reviewers

### CODE REVIEW

File: packages/circuits/circuits/qr_verify.circom [hunk 1]
```diff
@@ -5,11 +5,14 @@ include "./sha.circom";
 
 
 template QR_Verify(n, k, len) {
+    signal input signedMessage;
```
Comment (saleel) [inline packages/circuits/circuits/qr_verify.circom:8]: Is signedMessage the correct name? 
How about calling it `signal`?

File: packages/circuits/circuits/qr_verify.circom [hunk 2]
```diff
@@ -5,11 +5,14 @@ include "./sha.circom";
...
     signal input padded_message[len]; // private
     signal input message_len; // private 
     signal input signature[k]; //private
     signal input modulus[k]; //public
 
+    signal output messageSquare;
```
Comment (saleel) [inline packages/circuits/circuits/qr_verify.circom:14]: Does it need to be an output signal? The proof would change based on the input either way right?

File: packages/anon-aadhaar-pcd/src/types.ts [hunk 3]
```diff
@@ -1,4 +1,8 @@
-import { NumberArgument, StringArrayArgument } from '@pcd/pcd-types'
+import {
+  NumberArgument,
+  StringArgument,
```
Comment (Meyanis95) [inline packages/anon-aadhaar-pcd/src/types.ts:3]: I think we can remove this import if not used. core package will be updated in another PR.

Comment (saleel) [reply packages/anon-aadhaar-pcd/src/types.ts:3]: Yes, I will remove it another PR I am working on now related to refactor/cleanup

Comment (Meyanis95) [reply packages/anon-aadhaar-pcd/src/types.ts:3]: Which one? Because I already have some work to push, waiting for develop to be ready.
What is the scope of the refactor/cleanup?

File: packages/circuits/circuits/qr_verify.circom (hunk 2 above)
Comment (vuvoth) [reply packages/circuits/circuits/qr_verify.circom:14]: Yeah, you right! We don't need output signal.

File: packages/circuits/circuits/qr_verify.circom (hunk 1 above)
Comment (vuvoth) [reply packages/circuits/circuits/qr_verify.circom:8]: I think this is some message prover want to approve. So I refer signedMessage.

Comment (saleel) [reply packages/circuits/circuits/qr_verify.circom:8]: Yes, but its not "signed" by anyone technically right? Maybe we can simply call it `message`?

@Meyanis95 wdyt?

Comment (Meyanis95) [reply packages/circuits/circuits/qr_verify.circom:8]: Yes I'm advocating to just call it `signal`, as its role is to commit to a certain value (a signal) without actually "signing" it.

Comment (vuvoth) [reply packages/circuits/circuits/qr_verify.circom:8]: signal also the key work of circom, how about signalMessage.

Comment (Meyanis95) [reply packages/circuits/circuits/qr_verify.circom:8]: We can use signalHash then
https://github.com/semaphore-protocol/semaphore/blob/0602554b97f156a1c203330af2af91b5fbd886e3/packages/circuits/semaphore.circom#L53

File: packages/circuits/circuits/qr_verify.circom [hunk 4]
```diff
@@ -99,7 +101,11 @@ template QR_Verify(n, k, len) {
...
+
+    signal signalHashSquare; 
+
+    signalHashSquare <== signalHash * signalHash;
 }
 
 
-component main{public [modulus]} = QR_Verify(64, 32, 512 * 3);
\ No newline at end of file
+component main{public [signalHash]} = QR_Verify(64, 32, 512 * 3);
```
Comment (Meyanis95) [inline packages/circuits/circuits/qr_verify.circom:111]: `modulus` public declaration is missing here

**Comment by Meyanis95** [comment]
LGTM

**Review by Meyanis95 — APPROVED** [review]
